demos/common/cpp_gapi/utils_gapi/src/stream_source.cpp

```diff
@@ -16,14 +16,14 @@ void CommonCapSrc::preparation() {
     if (!tmp.data) {
         GAPI_Assert(false && "Couldn't grab the first frame");
     }
-    first = tmp.clone();
+    first = tmp;
```

Comment:
No. cv::Mat counts references, and dies only when there's no references

Reply:
I mean, it works that way :)

Reply:
Well, since you mess with `CommonCapSrc`
Without that `clone()` the whole `preparation()` body should be rewritten as
```cpp
    GAPI_Assert(first.empty());
    first = cap->read();
    GAPI_Assert(!first.data && "Couldn't grab the first frame");
```
and
```cpp
bool CommonCapSrc::pull(cv::gapi::wip::Data &data) {
    if (first.data) {
        data = std::move(first);
        first = cv::Mat{};
        return true;
    }
    cv::Mat frame = cap->read();
    data = frame;
    return !frame.data;
}
```

Reply:
that's a good point, and actually these changes almost don't depend on this `clone()` fix

I think this rewriting deserves a separate PR, this PR is too big already. @Wovchena  Do you agree with a separate PR being opened?

Reply:
Ok